src/pages/index.tsx

```diff
@@ -1,9 +1,165 @@
 import React from "react"
 
+import { Helmet } from "react-helmet"
+
+import Title from "decentraland-gatsby/dist/components/Text/Title"
+import useAuthContext from "decentraland-gatsby/dist/context/Auth/useAuthContext"
+import useAsyncTask from "decentraland-gatsby/dist/hooks/useAsyncTask"
+import useFormatMessage from "decentraland-gatsby/dist/hooks/useFormatMessage"
 import { Container } from "decentraland-ui/dist/components/Container/Container"
 
+import Places from "../api/Places"
+import ContainerWrapper from "../components/Layout/ContainerWrapper"
+import Navigation, { NavigationTab } from "../components/Layout/Navigation"
+import OverviewList from "../components/Layout/OverviewList"
+import { AggregatePlaceAttributes } from "../entities/Place/types"
+import {
+  usePlaceListMyFavorites,
+  usePlaceListPois,
+  usePlaceListPopular,
+  usePlaceListRecentlyUpdates,
+} from "../hooks/Place"
+import locations from "../modules/locations"
+
 import "./index.css"
 
-export default function IndexPage() {
-  return <Container></Container>
+const overviewOptions = { limit: 5, offset: 0 }
+
+export default function OverviewPage() {
+  const l = useFormatMessage()
+
+  const [account, accountState] = useAuthContext()
+  const [placeListLastUpdates, placeListLastUpdatesState] =
+    usePlaceListRecentlyUpdates(overviewOptions)
+  const [placeListPopular, placeListPopularState] =
+    usePlaceListPopular(overviewOptions)
+  const [placeListMyFavorites, placeListMyFavoritesState] =
+    usePlaceListMyFavorites(overviewOptions)
+  const [placeListPois, placeListPoisState] = usePlaceListPois(overviewOptions)
+
+  const [handlingFavorite, handleFavorite] = useAsyncTask(
+    async (event, place: AggregatePlaceAttributes) => {
+      if (account === null) {
+        accountState.select()
+      } else if (place) {
+        const favoritesResponse = await Places.get().updateFavorite(
+          place.id,
+          !place.user_favorite
+        )
+        if (favoritesResponse) {
+          if (placeListLastUpdates && placeListLastUpdates.length > 0) {
+            const placeListLastUpdatesUpdated = placeListLastUpdates?.map(
```

Comment:
can we generalize this some way

Reply:
Please look the new function updatePlaceInPlaceList